### PR TITLE
Properly localize numbers

### LINKED ISSUE #2547: [Bug]: Localized separators
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a bug report that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.

### Describe the bug

Localized decimal

### Expected Behavior

Use localized decimal separators ...

### Issue Labels

visual bug

### FreeTube Version

v0.17.1 Beta

### Operating System Version

Mac OS 10.11.6

### Installation Method

.dmg

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

_No response_

### Additional Information

_No response_

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
---
Properly localize numbers
---

**Pull Request Type**
Please select what type of pull request this is:
- [x] Bugfix

**Related issue**
closes #2547

**Description**
Number seperators (",", "." and " ") were not chosen based on locale and were always shown as ","

**Screenshots (if appropriate)**
image

**Testing (for code that is not small enough to be easily understandable)**
Used search with different locales (had to clear search to test)

**Desktop (please complete the following information):**
 - OS: Windows
 - OS Version: 11
 - FreeTube version: 0.17.1

### CODE REVIEW

**Review by absidue — APPROVED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I saw this https://github.com/FreeTubeApp/FreeTube/issues/853 issue from the same author. Is this harder to fix?

**Comment by ChunkyProgrammer** [comment]
> I saw this #853 issue from the same author. Is this harder to fix?

I haven't really looked into that issue

**Comment by Aiz0** [comment]
While working on #2559 I noticed that 'en' will be used to format Like and Dislike numbers if no locale in freetube is set.
Should 'en' be replaced with 'default' so system locales format is used for this?
```diff
diff --git a/src/renderer/components/watch-video-info/watch-video-info.js b/src/renderer/components/watch-video-info/watch-video-info.js
index 4746dc55..6cc98007 100644
--- a/src/renderer/components/watch-video-info/watch-video-info.js
+++ b/src/renderer/components/watch-video-info/watch-video-info.js
@@ -219,7 +219,7 @@ export default Vue.extend({
       }
 
       const locale = this.currentLocale.replace('_', '-')
-      return this.likeCount.toLocaleString([locale, 'en'])
+      return this.likeCount.toLocaleString([locale, 'default'])
     },
 
     parsedDislikeCount: function () {
@@ -228,7 +228,7 @@ export default Vue.extend({
       }
 
       const locale = this.currentLocale.replace('_', '-')
-      return this.dislikeCount.toLocaleString([locale, 'en'])
+      return this.dislikeCount.toLocaleString([locale, 'default'])
     },
 
     likePercentageRatio: function () {

```

**Comment by ChunkyProgrammer** [comment]
> While working on #2559 I noticed that 'en' will be used to format Like and Dislike numbers if no locale in freetube is set. Should 'en' be replaced with 'default' so system locales format is used for this?
> 
> ```diff
> diff --git a/src/renderer/components/watch-video-info/watch-video-info.js b/src/renderer/components/watch-video-info/watch-video-info.js
> index 4746dc55..6cc98007 100644
> --- a/src/renderer/components/watch-video-info/watch-video-info.js
> +++ b/src/renderer/components/watch-video-info/watch-video-info.js
> @@ -219,7 +219,7 @@ export default Vue.extend({
>        }
>  
>        const locale = this.currentLocale.replace('_', '-')
> -      return this.likeCount.toLocaleString([locale, 'en'])
> +      return this.likeCount.toLocaleString([locale, 'default'])
>      },
>  
>      parsedDislikeCount: function () {
> @@ -228,7 +228,7 @@ export default Vue.extend({
>        }
>  
>        const locale = this.currentLocale.replace('_', '-')
> -      return this.dislikeCount.toLocaleString([locale, 'en'])
> +      return this.dislikeCount.toLocaleString([locale, 'default'])
>      },
>  
>      likePercentageRatio: function () {
> ```

Sounds like a good idea to me, maybe keep 'en' but move it after default?

**Comment by absidue** [comment]
@ChunkyProgrammer You shouldn't need to specify default here as you are getting the locale properly through i18n, so there will always be a locale set, the other PR doesn't currently do that.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Screenshot (268)

Views says NaN?

File: src/renderer/components/watch-video-info/watch-video-info.js [hunk 1]
```diff
@@ -238,7 +239,7 @@ export default Vue.extend({
       if (this.hideVideoViews) {
         return null
       }
-      return this.viewCount.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',') + ` ${this.$t('Video.Views').toLowerCase()}`
+      return Intl.NumberFormat(this.currentLocale).format(this.viewCount.toString) + ` ${this.$t('Video.Views').toLowerCase()}`
```
Comment (Aiz0) [inline src/renderer/components/watch-video-info/watch-video-info.js:242]:
```suggestion
      return Intl.NumberFormat(this.currentLocale).format(this.viewCount) + ` ${this.$t('Video.Views').toLowerCase()}`
```

**Review by absidue — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by PikachuEXE — APPROVED** [review]

**Pull request merged**